### PR TITLE
Support volume cloning in volume and volume detail pages

### PR DESCRIPTION
**Pull request by a110605**
### What this PR does / why we need it
 - [x] In volume create  modal, users are able to create a volume with the data source pointing to a volume or a Longhorn snapshot
- [x] Users can click on a volume and select `clone volume` from that volume
- [x] Users can click on a snapshot and select `clone volume` from that snapshot
- [x] Users can click mutliple volumes to clone them, bulk action for volume clone 

### Issue
https://github.com/longhorn/longhorn/issues/8741

### Test Result

**Create volume with data source**

https://github.com/longhorn/longhorn-ui/assets/5744158/cb0788d7-0dbf-46ed-8372-eed6e2b97f9e

**Clone volume from existing volume**

https://github.com/longhorn/longhorn-ui/assets/5744158/23f6ae9f-4edd-4e2f-877a-ce65a3b2fdc7

**Clone volume from existing volume snapshot**

https://github.com/longhorn/longhorn-ui/assets/5744158/3e8b1669-d77c-4607-bef9-b95bfcdc984b

**Bulk volume cloning**

https://github.com/longhorn/longhorn-ui/assets/5744158/ba729788-b03f-4bce-8b4e-1e07eec94e5e

### CODE REVIEW

File: src/routes/volume/index.js [hunk 1]
```diff
@@ -1149,14 +1171,37 @@ class Volume extends React.Component {
...
+        })
+      },
+    }
+
     const addVolume = () => {
       dispatch({
         type: 'volume/showCreateVolumeModalBefore',
-      })
-      this.setState({
-        CreateVolumeGen() {
```
Comment (a110605) [inline src/routes/volume/index.js:1157]: unneeded setState

File: src/router.js [hunk 2]
```diff
@@ -114,7 +114,6 @@ const Routers = function ({ history, app }) {
             <Route exact path={`${path}instanceManager`} component={instanceManager} />
             <Route exact path={`${path}backingImage`} component={backingImage} />
             <Route exact path={`${path}recurringJob`} component={recurringJob} />
-            <Route exact path={`${path}recurringJob`} component={recurringJob} />
```
Comment (a110605) [inline src/router.js:117]: duplicated Route

File: src/models/volume.js [hunk 3]
```diff
@@ -212,6 +231,34 @@ export default {
         yield put({ type: 'changeTagsLoading', payload: { tagsLoading: false } })
       }
     },
+    *showCreateVolumeModalBefore({
+      payload,
+    }, { call, put, all }) {
+      yield put({ type: 'showCreateVolumeModal' })
+      // TODO: longhorn manager should have an API to get all volume's snapshots or add snapshotList array in GET /v1/volumes
```
Comment (a110605) [inline src/models/volume.js:238]: Hi @PhanLe1010 ,
In order to show all volumes snapshot in create volume modal. Current I sends mutliple POST requests to get each volume's snapshot list.

Screenshot 2024-06-24 at 6 22 22 PM

Is it possible to have new API `GET /v1/snapshots` to get all volume's snapshots like response
```
{
  vol_name: [ {...}, {...},...],
  vol1_name : [{..},{...}]
}
```

or put snapshot list data in `GET /v1/volumes` API response. 
That can save extra API call and save line 238 ~ 253
cc @derekbit

File: src/routes/volume/CloneVolume.js [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        </FormItem>
+        <FormItem label="Access Mode" hasFeedback {...formItemLayout}>
+          {getFieldDecorator('accessMode', {
+            initialValue: volume.accessMode || 'rwo',
+          })(<Select>
+            <Option key={'ReadWriteOnce'} value={'rwo'}>ReadWriteOnce</Option>
+            <Option key={'ReadWriteMany'} value={'rwx'}>ReadWriteMany</Option>
+          </Select>)}
+        </FormItem>
+        <FormItem label="Backing Image" hasFeedback {...formItemLayout}>
```
Comment (a110605) [inline src/routes/volume/CloneVolume.js:195]: Hi @PhanLe1010 or @ChanYiLin , is there any meaning that user choose other backing image while cloning the volume ? 
Can we disable or remove this field in volume clone modal ?

Screenshot 2024-06-27 at 12 06 24 PM

Comment (a110605) [reply src/routes/volume/CloneVolume.js:195]: After discussed with @ChanYiLin , we will disabled this backing image parameter while cloning volume.

**Comment by PhanLe1010** [comment]
In gereral, the PR LGTM

**Comment by a110605** [comment]
> In gereral, the functionalities LGTM
> 
> Disclaimer: I am not be able to review the actual code

No problem :) Just make sure the UI change matches your expectation.

File: src/models/volume.js (hunk 3 above)
Comment (a110605) [reply src/models/volume.js]: Thanks for advice. That make sense. Modified in 2fed9e8, the behavior change like below screen video

https://github.com/longhorn/longhorn-ui/assets/5744158/bfe78da3-ac14-4ca1-9453-7755388b06ab

**Review by torchiaf — APPROVED** [review]
LGTM

**Pull request merged**